storage/src/backend/connection.rs

```diff
@@ -327,6 +384,86 @@ impl Connection {
             }
         }
 
+        let current_mirror = self.mirror_state.current.load();
+
+        if let Some(mirror) = current_mirror.as_ref() {
+            let data_cloned: Option<ReqBody<R>> = match data.as_ref() {
+                Some(ReqBody::Form(form)) => Some(ReqBody::Form(form.clone())),
+                Some(ReqBody::Buf(buf)) => Some(ReqBody::Buf(buf.clone())),
+                _ => None,
+            };
+
+            let mirror_host = Url::parse(&mirror.config.host).map_err(ConnectionError::Url)?;
+
+            let mut current_url = Url::parse(url).map_err(ConnectionError::Url)?;
+
+            current_url
+                .set_scheme(mirror_host.scheme())
+                .map_err(|_| ConnectionError::Scheme)?;
+            current_url
+                .set_host(mirror_host.host_str())
+                .map_err(|_| ConnectionError::Host)?;
+            current_url
+                .set_port(mirror_host.port())
+                .map_err(|_| ConnectionError::Port)?;
+
+            for (key, value) in mirror.config.headers.iter() {
+                headers.insert(
+                    HeaderName::from_str(key).unwrap(),
+                    HeaderValue::from_str(value).unwrap(),
+                );
+            }
+
+            let result = self.call_inner(
+                &mirror.client,
+                method.clone(),
+                current_url.to_string().as_str(),
+                &query,
+                data_cloned,
+                headers,
+                catch_status,
+                false,
+            );
+
+            match result {
+                Ok(resp) => {
+                    if resp.status() < StatusCode::INTERNAL_SERVER_ERROR {
+                        return Ok(resp);
+                    }
+                }
+                Err(err) => {
+                    warn!(
+                        "request mirror failed, mirror: {:?},  error: {:?}",
+                        mirror, err
+                    );
+                    mirror.fail_times.fetch_add(1, Ordering::Relaxed);
+
+                    if mirror.fail_times.load(Ordering::Relaxed) >= mirror.fail_limit {
+                        error!(
+                            "reach to fail limit {}, disable mirror: {:?}",
+                            mirror.fail_limit, mirror
+                        );
+                        mirror.status.store(false, Ordering::Relaxed);
+
+                        let mut idx = 0;
+                        loop {
+                            if idx == self.mirror_state.mirrors.len() {
+                                break None;
+                            }
+                            let m = &self.mirror_state.mirrors[idx];
+                            if m.status.load(Ordering::Relaxed) {
+                                break Some(m);
```

Comment:
Maybe give a warning to notify user that we have changed the mirror server?

Reply:
Yes, Fixed.